logstash-core/src/main/java/org/logstash/log/RubyBasicObjectSerializer.java

```diff
@@ -0,0 +1,116 @@
+
+    private static final long serialVersionUID = -5557562960691452054L;
+    private static final Logger LOGGER = LogManager.getLogger(RubyBasicObjectSerializer.class);
+    private static final String METHOD_INSPECT = "inspect";
+    private static final String METHOD_TO_STRING = "to_s";
+
+    public RubyBasicObjectSerializer() {
+        super(RubyBasicObject.class);
+    }
+
+    @Override
+    public void serialize(final RubyBasicObject value, final JsonGenerator gen, final SerializerProvider provider) throws IOException {
+        final Optional<JsonSerializer<Object>> serializer = findTypeSerializer(value, provider);
+        if (serializer.isPresent()) {
+            try {
+                serializer.get().serialize(value, gen, provider);
+                return;
+            } catch (IOException e) {
+                LOGGER.debug("Failed to serialize value type {} using default serializer {}", value.getClass(), serializer.get().getClass(), e);
+            }
+        }
+
+        if (isCustomInspectMethodDefined(value)) {
+            try {
+                gen.writeString(value.callMethod(METHOD_INSPECT).asJavaString());
+                return;
+            } catch (Exception e) {
+                LOGGER.debug("Failed to serialize value type {} using the custom `inspect` method", value.getMetaClass(), e);
+            }
+        }
+
+        try {
+            gen.writeString(value.callMethod(METHOD_TO_STRING).asJavaString());
+            return;
+        } catch (Exception e) {
+            LOGGER.debug("Failed to serialize value type {} using `to_s` method", value.getMetaClass(), e);
+        }
+
+        try {
+            gen.writeString(value.to_s().asJavaString());
+        } catch (Exception e) {
+            LOGGER.debug("Failed to serialize value type {} using `RubyBasicObject#to_s()` method", value.getMetaClass(), e);
+            gen.writeString(value.anyToString().asJavaString());
+        }
+    }
+
+    private Optional<JsonSerializer<Object>> findTypeSerializer(final RubyBasicObject value, final SerializerProvider provider) {
+        if (ClassUtil.isCollectionMapOrArray(value.getClass())) {
+            try {
+                // Delegates the serialization to the Jackson's default serializers, which might
+                // end up using this serializer for its elements.
+                return Optional.ofNullable(provider.findTypedValueSerializer(value.getJavaClass(), false, null));
+            } catch (JsonMappingException e) {
+                // Ignored
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private boolean isCustomInspectMethodDefined(final RubyBasicObject value) {
+        try {
+            final Object candidate = value.method(RubyString.newString(RubyUtil.RUBY, METHOD_INSPECT));
+            return candidate instanceof RubyMethod && ((RubyMethod) candidate).owner(RubyUtil.RUBY.getCurrentContext()).toString().toLowerCase().startsWith("logstash");
```

Comment:
Same as above, do we know if this is costly performance wise when there's a lot of log entries? If necessary we could consider some memoization/caching strategy.